node-data/src/ledger/header.rs

```diff
@@ -20,11 +20,11 @@ pub struct Header {
     pub event_hash: Hash,
     pub generator_bls_pubkey: PublicKeyBytes,
     pub txroot: Hash,
-    pub faultroot: Hash,
     pub gas_limit: u64,
     pub iteration: u8,
     pub prev_block_cert: Attestation,
     pub failed_iterations: IterationsInfo,
+    pub faults: Vec<Fault>,
```

Comment:
I think this list belongs to the block. 
If there's no strong reason not do to it, I'd keep `faultroot` here.

Reply:
I did it to multiple reason:
1. With empty faults, the fault root increases the block size by 32bytes. A `Vec<Faults>` inreases only by 4 bytes
2. It currently simplifies the validations, because I moved the verify_faults into the verify_header

If what concerns you is the fact that the vec is unbounded, we can put a cap.

Anyway, we can handle this while tackling https://github.com/dusk-network/rusk/issues/1969 .
We can even decide to put faults into transactions as SlashTransaction

Reply:
The problem with tackling this with #1969 is that it's potentially scheduled for after mainnet, but this one is a breaking change. 
To be honest, I think 32 bytes is acceptable. We waste much more with the Attestation field, which is not needed in the block. 
Having slash transactions would be also fine IMO.

As for validation, let's discuss it offline.

Reply:
Moved back to fault_tree